### ISSUE TITLE
[winget] Unable to install the new alpha

### ISSUE THREAD

**Issue by lesquoyb** [gama-platform/gama.old#3917]
**Describe the bug**
Though the latest version has been merged successfully into the winget repository, when running `winget install GamaPlatform.GamaAlpha` it tries to install the previous alpha (`GAMA_1.9.3_Windows_with_JDK_09.10.23_89d4463.exe`) which in addition doesn't work because the link to it is broken.
I think I understood the problem: both versions are available in the winget repository, and the version numbers are the same appart from the end that is a piece of the version's hash. So the previous one is called [1.9.3-89d4463](https://github.com/microsoft/winget-pkgs/tree/master/manifests/g/GamaPlatform/GamaAlpha/1.9.3-89d4463) and latest one is called [1.9.3-68338e3](https://github.com/microsoft/winget-pkgs/tree/master/manifests/g/GamaPlatform/GamaAlpha/1.9.3-68338e3), which leads to winget picking the previous one as the most up to date (maybe?).
So to get out of this problem I think we should probably only have one version in the alpha, as the link will be broken anyway for the previous versions. Another way out would be to include the date/datetime into the version number.
@WoodenMaiden , @RoiArthurB what are your thoughts ?

**Comment by WoodenMaiden** [comment]
I'm pretty sure I know what happened

The [publish to winget action](https://github.com/vedantmgoyal2009/winget-releaser) has an argument named `max-versions-to-keep` to do just that by sending another PR to delete previous versions AFTER the PR to send the new one.

When I ran the deployment yesterday the action gave me this error:
```
>
Cleaning up previous stale branches of merged pull requests on the fork...
undefined:1
A server error has occurred
^

SyntaxError: Unexpected token 'A', "A server e"... is not valid JSON
    at JSON.parse (<anonymous>)
    at parseJSONFromBytes (node:internal/deps/undici/undici:6662:19)
    at successSteps (node:internal/deps/undici/undici:6636:27)
    at node:internal/deps/undici/undici:1236:60
    at node:internal/process/task_queues:140:7
    at AsyncResource.runInAsyncScope (node:async_hooks:206:9)
    at AsyncResource.runMicrotask (node:internal/process/task_queues:137:8)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
```

It happened on the PR creating the new version, meaning that the action ended too early and didn't sent the PR for deletion.

Maybe we should replace the commit hash in the version by a counter (eg.`1.9.3-1` `1.9.3-2` `1.9.3-3`...), until the problem is fixed

**Comment by WoodenMaiden** [comment]
For now I will manually delete the old version

**Comment by lesquoyb** [comment]
Yes, the counter is a good idea :)
Also the manual deletion worked I managed to download the correct version, still keeping the issue open for the next one

**Comment by RoiArthurB** [comment]
@WoodenMaiden @lesquoyb Didn't follow if this issue is fixed and/or this needs testing/work (if so, any of you is on it) ? :)

**Comment by lesquoyb** [comment]
Currently working as expected yes, I just run a gh job to produce a new release to check it's updated as it should, I'll close when done.

**Comment by lesquoyb** [comment]
There may be an error there: the publish to windows part was completely skipped. I know we talked about it some time ago @WoodenMaiden, but I can't remember why it is this way, can you clarify ?
In my memory it was only the big releases that were supposed to be triggered manually and the alpha would be pushed automatically or something like this.
Anyway pushing a new release on gh did break it again as now winget doesn't have the link to the newest alpha.

**Comment by WoodenMaiden** [comment]
The software used by the action kept throwing errors, so I shut it down and did the updates manually until the error is fixed, however the [issue is still open](https://github.com/vedantmgoyal2009/winget-releaser/issues/136#issuecomment-1753807641). I'll try to reactivate and run the windows release, if this still not work I'll do differently.

**Comment by WoodenMaiden** [comment]
https://github.com/microsoft/winget-pkgs/pull/128265
Wow, that worked ?

**Comment by lesquoyb** [comment]
reopening because the winget install still points to the previous build for some reason:
![image](https://github.com/gama-platform/gama/assets/6374469/6fb89260-fa6e-4528-a0b4-17d8fe106765)

**Comment by RoiArthurB** [comment]
@lesquoyb last accepted yesterday was pointing on the good new URL https://github.com/microsoft/winget-pkgs/pull/128265/files ...

Can you update or something your winget local db (if it's something making sense with winget)?

**Comment by lesquoyb** [comment]
No local db that I'm aware of, it seems to download the list of packages everytime. There are two versions in the [repo](https://github.com/microsoft/winget-pkgs/tree/master/manifests/g/GamaPlatform/GamaAlpha), and my guess is that the previous one is picked before the latest because of its name

**Comment by lesquoyb** [comment]
@WoodenMaiden could you do your magic to remove the old [1.9.3-ca4a0ca](https://github.com/microsoft/winget-pkgs/tree/master/manifests/g/GamaPlatform/GamaAlpha/1.9.3-ca4a0ca) from winget repo and I think it should fix the last issue now that the counter works ?

**Comment by WoodenMaiden** [comment]
just wait on this PR to be approved

https://github.com/microsoft/winget-pkgs/pull/129123

**Comment by lesquoyb** [comment]
seems like there's still an issue, I tried to build and deploy but the winget part [failed](https://github.com/gama-platform/gama/actions/runs/7097656164/job/19319317628)

**Comment by WoodenMaiden** [comment]
It's fine if you read the logs you can see it has been published (https://github.com/microsoft/winget-pkgs/pull/129243), you just have to wait for the PR to me merged. 
The failing step has been fixed at 1c38cb5

**Comment by lesquoyb** [comment]
perfect, finally closing then 🥳
